src/itensor.jl

```diff
@@ -67,33 +66,30 @@ NDTensors.Dense{Float64,Array{Float64,1}}
   1.2579101497658178  -1.3559959053693322
 ```
 """
-mutable struct ITensor{N}
+mutable struct ITensor
   store::TensorStorage
-  inds::IndexSet{N}
+  inds::IndexSet
 
   # TODO: check that the storage is consistent with the 
   # indices (possibly only in debug mode);
   """
-      ITensor{N}(is::IndexSet{N}, st::TensorStorage)
+      ITensor(is::IndexSet, st::TensorStorage)
 
   This is an internal constructor for an ITensor where the ITensor stores a view of the `NDTensors.TensorStorage`.
   """
-  ITensor{N}(is, st::TensorStorage) where {N} = new{N}(st, is)
+  ITensor(is, st::TensorStorage) = new(st, is)
 
-  ITensor{Any}(is, st::Empty) = new{Any}(st, is)
+  ITensor(is, st::Empty) = new(st, is)
```

Comment:
This constructor is covered by the one above now is isn't necessary.

The meaning of an ITensor with an order of `Any` was a special case where you wanted an ITensor to an initialize a loop where you are adding onto the ITensor in a loop but didn't know what the order of the ITensor would be:
```julia
A = emptyITensor(Any)
for n in 1:10
  A += X * Y # Some expression where you don't know the order of the result
end
```
I would have to think about the fate of this syntax with the new design. The value `Any` was stored in the type parameter, so the ITensor was of type `ITensor{Any}` and that's how we determined it could be added to an ITensor with any number of indices. This isn't possible with the new design. We could replace this with just:
```julia
A = emptyITensor()
for n in 1:10
  A += X * Y # Some expression where you don't know the order of the result
end
```
and/or
```julia
A = ITensor(0)
for n in 1:10
  A += X * Y # Some expression where you don't know the order of the result
end
```
I think we were hesitant about those versions since it seemed like a slippery slope to allow adding a scalar ITensor to a nonscalar ITensor (it would imply that we should also allow `ITensor(2) + ITensor(i, j)` which was deemed unsavory). @emstoudenmire, any opinion on this? I think the version `A = emptyITensor()` above is pretty unambiguous so we could just use that for this use case.

Reply:
Yes, my vote is that `emptyITensor()` would now replace `emptyITensor(Any)` as the way to get the behavior where you can add any other ITensor to it and it will act as a "universal additive zero". Also we could consider allowing setting the elements of an empty ITensor, and perhaps make the constructor `ITensor()` do the same thing as `emptyITensor()` but those are later considerations.

We definitely don't want `ITensor(x)` for some scalar x to act as an `emptyITensor()`, I think even including the case x=0.

Reply:
Sounds good. Agreed that we can consider having `ITensor()` act like `emptyITensor()` in the future.